src/hotspot/cpu/x86/stubGenerator_x86_64_adler.cpp

```diff
@@ -132,28 +139,72 @@ address StubGenerator::generate_updateBytesAdler32() {
   __ jcc(Assembler::aboveEqual, SKIP_LOOP_1A);
 
   __ align32();
-  __ bind(SLOOP1A);
-  __ vbroadcastf128(ydata, Address(data, 0), Assembler::AVX_256bit);
-  __ addptr(data, CHUNKSIZE);
-  __ vpshufb(ydata0, ydata, yshuf0, Assembler::AVX_256bit);
-  __ vpaddd(ya, ya, ydata0, Assembler::AVX_256bit);
-  __ vpaddd(yb, yb, ya, Assembler::AVX_256bit);
-  __ vpshufb(ydata1, ydata, yshuf1, Assembler::AVX_256bit);
-  __ vpaddd(ya, ya, ydata1, Assembler::AVX_256bit);
-  __ vpaddd(yb, yb, ya, Assembler::AVX_256bit);
-  __ cmpptr(data, end);
-  __ jcc(Assembler::below, SLOOP1A);
+  if (VM_Version::supports_avx512vl()) {
+    __ cmpl(s, VM_Version::avx3_threshold());
+    __ jcc(Assembler::belowEqual, SPRELOOP1A_AVX2);
+    // AVX2 performs better for smaller inputs because of leaner post loop reduction sequence..
+    __ cmpl(s, 128);
+    __ jcc(Assembler::belowEqual, SPRELOOP1A_AVX2);
```

Comment:
These two compares can be merged into one compare with larger of avx3_threshold() or 128.